hooks/pre/31-upgrade-puppet.rb

```diff
@@ -0,0 +1,54 @@
+def stop_services                                                                                                                                                                                                                                     [0/1892]
+  Kafo::Helpers.execute('katello-service stop')
+end
+
+def upgrade_puppet_pkg
+  Kafo::Helpers.execute('yum remove -y puppet-server')
+  Kafo::Helpers.execute('yum install -y puppetserver')
+  Kafo::Helpers.execute('yum install -y puppet-agent')
+end
+
+def populate_p4_environments
+  Kafo::Helpers.execute('cp -rfp /etc/puppet/environments/* /etc/puppetlabs/code/environments')
+end
+
+def copy_ssl_certs
```

Comment:
This doesn't just copy the certs, also the (answer) cache. Maybe copy_data would be a better name?